lib/tests/formulae.rb

```diff
@@ -294,10 +294,14 @@ def formula!(formula_name, args:)
 
         # Don't care about e.g. bottle failures for dependencies.
         test "brew", "install", "--only-dependencies", *install_args,
-             env:  { "HOMEBREW_DEVELOPER" => nil }
+             env: { "HOMEBREW_DEVELOPER" => nil }
 
+        env = {}
+        env["HOMEBREW_GIT_PATH"] = nil if deps.any? do |d|
+          d.name == "git" && (!d.test? || d.build?)
+        end
```

Comment:
Could this get pulled into a function that takes `deps` as an argument and returns `env`? It's used in 4 places.